src/api/app/controllers/webui/request_controller.rb

```diff
@@ -402,6 +384,25 @@ def accept_request
     end
   end
 
+  def prepare_bsrequest(action, params)
+    req = BsRequest.new(state: 'new', description: params[:description])
+    params.delete :description
+    klass_to_call = case action
+                    when :set_bugowner
+                      BsRequestActionSetBugowner
+                    when :delete
+                      BsRequestActionDelete
+                    when :add_role
+                      BsRequestActionAddRole
+                    when :change_devel
+                      BsRequestActionChangeDevel
+                    else
+                      raise APIException, 'method not found'
```

Comment:
BsRequestAction not found or even BsRequestAction '#{action}' not found

Reply:
are we rendering this snippet? Otherwise I would avoid to return some user controlled value

Reply:
It will be rendered in the error xml

Reply:
so because it's being rendered, I would prefer to have it generic and not under user control

Reply:
There usually is an errorcode (as string) and of course the http status that can be used by bots / scripts.

It often helps to know why some request failed, eg. was the input wrong or not received at all? That's not clear if you just tell users 'unknown action'